### DISCUSSION TITLE
Extract HOMM2 Resources Problem

### DISCUSSION THREAD

**Discussion by adamoftheburti** [ihhub/fheroes2#5810] [Q&A]
I have a legally purchased version of HOMM2 and am trying to run "extract_homm2_resources.sh" on my Mac. I keep getting "Unable to find the HoMM2 directory. Installation aborted." What do I do? I'm not being prompted to enter the path, it just goes straight to that error.

I'm running macOS Monterey on an Intel chip.

**Comment by zenseii** [comment]
Hi @adamoftheburti !

You'll have to excuse the late reply. Many other users have similar issues with Mac.
I take it you have been able to actually install the original Heroes 2 game then? Which version of Heroes 2 do you have? Digital or physical?

**Reply by adamoftheburti** [reply]
@zenseii No worries—you're having to keep up with a lot! I have the digital version of Heroes 2 and have been able to install it on windows but not mac.

**Reply by zenseii** [reply]
@Oleg-Dereventz, I'd like to ask your help here if you have the time. I believe you are the most familiar with setting up on Mac?

**Reply by oleg-derevenetz** [reply]
Hi all,

I was unable to reproduce this issue. That's how this script works for me:

```
oleg@MacBook-Pro homm2 % ./extract_homm2_resources.sh 
This script will extract and copy game resources from the original distribution of Heroes of Might and Magic II

[1/3] determining the destination directory

Destination directory: ../..

[2/3] determining the HoMM2 directory

Please enter the full path to the HoMM2 directory (e.g. /home/user/GOG Games/HoMM 2 Gold): 
```

Thus, it requests the full path to the original HoMM2 distribution. The only cause when it will not request it is if this path is already specified as a command-line argument:

```
oleg@MacBook-Pro homm2 % ./extract_homm2_resources.sh /home/oleg/homm2
This script will extract and copy game resources from the original distribution of Heroes of Might and Magic II

[1/3] determining the destination directory

Destination directory: ../..

[2/3] determining the HoMM2 directory

Unable to find the HoMM2 directory. Installation aborted.
```

@adamoftheburti could you please provide the exact script's command line (with all arguments) as well as its output, just like I did above, so that I can try to reproduce the issue?

P.S. Please note that this script is designed to use `bash`, not `zsh` that is now the default shell on macOS. But it has the corresponding shebang, so, in theory, it cannot be accidentally launched via `zsh` - only if you use `zsh` intentionally, like `zsh extract_homm2_resources.sh`.
